### PR TITLE
Mes 1721 send logs

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
## Description and relevant Jira numbers

Sending logs every minute.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/modules/logs/logs.actions.ts [hunk 1]
```diff
@@ -1,12 +1,38 @@
 import { Action } from '@ngrx/store';
 import { Log } from '../../shared/models/log.model';
 
-export const SAVE_LOG = 'Save Log';
+export const SAVE_LOG = '[Anywhere] Save Log';
```
Comment (Cavalier07) [inline src/modules/logs/logs.actions.ts]: How about [GLOBAL] or [SHARED]? I don't really like Anywhere

File: src/modules/logs/logs.effects.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { Effect, Actions, ofType } from '@ngrx/effects';
+
+import { switchMap, map, catchError, mapTo, withLatestFrom } from 'rxjs/operators';
+import { of } from 'rxjs/observable/of';
+import { interval } from 'rxjs/observable/interval';
+import { Store, select } from '@ngrx/store';
```
Comment (Cavalier07) [inline src/modules/logs/logs.effects.ts]: Could this import be moved to the top group of imports.

File: src/providers/url/url.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export class UrlProvider {
   }
 
   getLoggingServiceUrl(): any {
-    return this.appConfigProvider.getAppConfig().loggingUrl;
+    return this.appConfigProvider.getAppConfig().logging.url;
```
Comment (Cavalier07) [inline src/providers/url/url.ts]: What are your thoughts about changing this to getLogsServiceUrl . It seems like everything else has moved from logging to logs

File: src/modules/logs/logs.effects.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  @Effect()
+  sendLogsEffect$ = this.actions$.pipe(
+    ofType(logsActions.SEND_LOGS),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getLogsState),
+      ),
+    ),
+    switchMap(([action, logs]) => {
+      console.log(logs);
```
Comment (Cavalier07) [inline src/modules/logs/logs.effects.ts]: Console Log

File: src/providers/logs/logs.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { Log } from '../../shared/models/log.model';
+import { UrlProvider } from '../url/url';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class LogsProvider {
+
+  constructor(public http: HttpClient, public urlProvider: UrlProvider) {}
+
+  public log = (log: Log): Observable<Object> => {
```
Comment (Cavalier07) [inline src/providers/logs/logs.ts]: This function is only used in the test now as the effect calls logMultiple() - is there a use where we would only send one log to the service , bypassing the store.

File: src/providers/logs/logs.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    return this.logMultiple([log]);
+  }
+
+  public logMultiple = (logs : Log[]): Observable<Object> => {
+    const loggingServiceUrl = this.urlProvider.getLoggingServiceUrl();
```
Comment (Cavalier07) [inline src/providers/logs/logs.ts]: Is there a better return type you can use instead of Object

**Comment by johnmorleybjss** [comment]
Should there be code in a reducer that cleans out the logs once they have been submitted?

**Comment by szabolcs-keresztes** [comment]
@johnmorleybjss yes, there is going to be functionality in place which deletes the already sent logs, coming as part of a next PR.

File: src/modules/logs/logs.actions.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/modules/logs/logs.actions.ts]: We can agree on *[GLOBAL]* for these kind of actions, I like it.

File: src/modules/logs/logs.effects.ts (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/modules/logs/logs.effects.ts]: sure

File: src/providers/url/url.ts (hunk 3 above)
Comment (szabolcs-keresztes) [reply src/providers/url/url.ts]: Also the microservice is called `mes-logs-service` so ye it totally makes sense. But that would require an additional PR in the configuration microservice. Do you think that it's worth it?

File: src/providers/logs/logs.ts (hunk 5 above)
Comment (szabolcs-keresztes) [reply src/providers/logs/logs.ts]: I think we are going to have the SEND_LOGS action sending logs for us. And that works with multiple even if there is one log in the array. So I'll delete the `log` function. And I also going to rename `logMultiple` to `sendLogs` function.

File: src/providers/url/url.ts (hunk 3 above)
Comment (szabolcs-keresztes) [reply src/providers/url/url.ts]: Changed it and refactored the code so that we use `logs` everywhere (Global hah).

**Pull request merged**